clang/lib/Sema/SemaExpr.cpp

```diff
@@ -1114,8 +1114,6 @@ static bool handleIntegerToComplexFloatConversion(Sema &S, ExprResult &IntExpr,
   if (IntTy->isIntegerType()) {
     QualType fpTy = ComplexTy->castAs<ComplexType>()->getElementType();
     IntExpr = S.ImpCastExprToType(IntExpr.get(), fpTy, CK_IntegralToFloating);
-    IntExpr = S.ImpCastExprToType(IntExpr.get(), ComplexTy,
-                                  CK_FloatingRealToComplex);
```

Comment:
This change makes the comment on this method and the comment in `handleComplexConversion` inaccurate; it would be a good idea to update those comments to indicate what is actually happening here.

Reply:
That’s a good point.

Reply:
I’ve updated the comments and also renamed `handleIntegerToComplexFloatConversion()` to `handleComplexIntegerToFloatConversion()` since the result is no longer always a complex float.